### DISCUSSION TITLE
error: Server received SIGSEGV signal

### DISCUSSION THREAD

**Discussion by meowwwwww95** [UPPAALModelChecker/UPPAAL-Meta#289] [Q&A]
When I tried to simulate my models in Uppaal 5.0.0, I met Error "Server received SIGSEGV signal". The models and error trace are added at the end. 
Actually, the models are able to simulate successfully without model Rule1 while if Rule1 is uploaded, the error occurs no matter whether Rule2 is uploaded. I have no idea what happens.

[error_case.zip](https://github.com/user-attachments/files/17090107/error_case.zip)

The error trace graph in simulator is followed:
![image](https://github.com/user-attachments/assets/673ad5a7-25b6-4ef4-94a1-cdfacfd6ff56)

**Comment by mikucionisaau** [comment]
Very strange, the engine is just crashing by attempting to execute that transition.
Uppaal 5.1.0-beta5 is also affected.

**Reply by meowwwwww95** [reply]
I make a further test. After deleting all synchronizations in humidity template, the simulator executes successfully, even if the result isn't what I expected. I hope that this test will assist your progress.

**Reply by mikucionisaau** [reply]
Here is the minimal example triggering the issue:
![image](https://github.com/user-attachments/assets/74e4dbf4-88c1-4c95-a7b8-b86fef49f171)
The issue relates to urgent broadcast synchronization to two receivers with zero delay (over committed location). And the issue is caused by a missing negation in `if` condition, deep in the evaluation of possible edges, oh well.

It does not affect Verifier: the issue is limited to the computation of the list of transitions in the interactive Concrete Simulator, where it computes *all possible* delays and synchronization successors over broadcast channel (whereas Verifier's simulator just computes one successor).

Unfortunately I do not have a workaround for this, so this needs another beta release.
